V2rayNG/app/src/main/kotlin/com/v2ray/ang/service/V2RayVpnService.kt

```diff
@@ -200,9 +200,12 @@ class V2RayVpnService : VpnService(), ServiceControl {
 //        val emptyInfo = VpnNetworkInfo()
 //        val info = loadVpnNetworkInfo(configName, emptyInfo)!! + (lastNetworkInfo ?: emptyInfo)
 //        saveVpnNetworkInfo(configName, info)
-        if (Build.VERSION.SDK_INT >= Build.VERSION_CODES.P && listeningForDefaultNetwork) {
-            connectivity.unregisterNetworkCallback(defaultNetworkCallback)
-            listeningForDefaultNetwork = false
+        if (Build.VERSION.SDK_INT >= Build.VERSION_CODES.P) {
+            try {
+                connectivity.unregisterNetworkCallback(defaultNetworkCallback)
```

Comment:
I feel like we can simply check null for defaultNetworkCallback then we don't need try catch?

Reply:
catch is needed for avoiding crash. the reason of IOException is something i do not care (actually Android has too much addition like this). exception at registerDefaultNetwork could be simply ignored even now.

Reply:
Fine, we don't care for unregister(). But we should care for register(). We have this code for a reason and some users are depending on it. We should at least add a log for exception. See more details
https://github.com/shadowsocks/shadowsocks-android/commit/3a8f863bd1f0deed081de082a674020368f30319